lib/sycamore/sycamore/query/execution/aggregation.py

```diff
@@ -0,0 +1,90 @@
+from collections.abc import Callable
+from typing import Any, Optional
+
+from ray.data import Dataset
+from ray.data.aggregate import AggregateFn
+from ray.data.grouped_data import GroupedData
+
+from sycamore.data import Document
+from sycamore.docset import DocSet
+from sycamore.plan_nodes import Scan
+
+
+# Todo support multiple fields
+def make_aggregation_map_fn(field: str, default_val: Any):
+    def ray_callable(input_dict: dict[str, Any]) -> dict[str, Any]:
+        doc = Document.from_row(input_dict)
+        if field not in doc.properties:
+            agg_val = default_val
+        else:
+            agg_val = doc.properties[field]
+
+        new_doc = doc.to_row()
+        new_doc[field] = agg_val
+
+        return new_doc
+
+    return ray_callable
+
+
+def accumulate_row(doc, doc_bytes, fn):
+    # print(f"In accumulate_row: {doc} {doc_bytes}")
```

Comment:
There are a bunch of commented out prints in this file that I would probably remove.